locator/abstract_replication_strategy.cc

```diff
@@ -696,27 +696,34 @@ future<global_vnode_effective_replication_map> make_global_effective_replication
 
 } // namespace locator
 
-std::ostream& operator<<(std::ostream& os, locator::replication_strategy_type t) {
+auto fmt::formatter<locator::replication_strategy_type>::format(locator::replication_strategy_type t,
+                                                                fmt::format_context& ctx) const -> decltype(ctx.out()) {
+    std::string_view name;
     switch (t) {
-    case locator::replication_strategy_type::simple:
-        return os << "simple";
-    case locator::replication_strategy_type::local:
-        return os << "local";
-    case locator::replication_strategy_type::network_topology:
-        return os << "network_topology";
-    case locator::replication_strategy_type::everywhere_topology:
-        return os << "everywhere_topology";
+    using enum locator::replication_strategy_type;
+    case simple:
+        name = "simple";
```

Comment:
I think that these names from `db/config.cc` are better:
```
std::unordered_map<sstring, locator::replication_strategy_type> db::replication_strategy_restriction_t::map() {
    return {{"SimpleStrategy", locator::replication_strategy_type::simple},
            {"LocalStrategy", locator::replication_strategy_type::local},
            {"NetworkTopologyStrategy", locator::replication_strategy_type::network_topology},
            {"EverywhereStrategy", locator::replication_strategy_type::everywhere_topology}};
}
```
And I also wonder if this formatter can be used to replace the mapping defined in `db/config`, as we probably don't want to keep 2 different type-string mappings for the same type.

Reply:
@ptrsmrn hi Piotr, the goal of this change is to replace the existing operator<< with an equivalent `fmt::formatter` specialization. so i'm inclined to preserve the existing behavior. that's also why i claim this change as a cleanup. we can always create a follow up to use the names from `db::replication_strategy_restriction_t::map()`. yeah, i am aware of it. and i also want to minimize the repeatings. let's do this change in a follow-up.

Reply:
Sure, this PR alone is also an improvement.